open-telemetry/opamp-go
# Add OpAMP connection settings capability to the example

This change demonstrates OpAMP Connection Setting Offer Flow from OpAMP spec (see https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#opamp-connection-setting-offer-flow)

- The example agent and server now use TLS for OpAMP connection.
- The server's certificate is signed by a CA authority. The CA authority's certificate is self-generated (can be re-generated using examples/certs/generate.sh).
- The server may generate and offer a client-side certificate to the agent. The generated certificate is signed by the same CA.
- The client can accept the certificate and use it for future connections.
- Server UI now allows the user to generate a certificate to offer to the agent (for the first time or to rotate). The flows are described in the OpAMP spec, see https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#trust-on-first-use

 WARNING: all included certificate files are examples only and MUST NOT
 be used in production.

 How to test:
 1. Run the server.
 2. Run the agent.
 3. Go to http://localhost:4321/
 4. Find the connected agent in the list and click it
 5. Scroll down and click "Accept and Offer Client Certificate" button.
![image](https://github.com/open-telemetry/opamp-go/assets/4194920/e485141c-1a9d-43db-934f-e612fd80ea13)
6. The certificate should be generated, sent to agent, accepted and agent will reconnect and the details should be shown:
![image](https://github.com/open-telemetry/opamp-go/assets/4194920/97bd7045-912b-4aea-bd7a-8539cbd85494)
